xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.h

```diff
@@ -104,6 +105,7 @@ class CProcessorHD : public ID3DResource
   D3D11_VIDEO_PROCESSOR_RATE_CONVERSION_CAPS m_rateCaps;
   D3D11_TEXTURE2D_DESC m_texDesc;
   PROCESSOR_VIEW_TYPE m_eViewType;
+  std::vector<AVPixelFormat> m_formats;
```

Comment:
this is static information you can collect at system start. it won't change during runtime, hence no need to overcomplicate startup of playback. take this as an example: https://github.com/FernetMenta/kodi-agile/blob/videobuffer/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp#L1195
this is called at startup by windowing. you could do the same with your renderer